src/tile.cpp

```diff
@@ -565,17 +565,17 @@ ReturnValue Tile::queryAdd(int32_t, const Thing& thing, uint32_t, uint32_t flags
 		}
 
 		const CreatureVector* creatures = getCreatures();
-		if (const Player* player = creature->getPlayer()) {
+		if (auto player = creature->getPlayer()) {
 			if (creatures && !creatures->empty() && !hasBitSet(FLAG_IGNOREBLOCKCREATURE, flags) &&
 			    !player->isAccessPlayer()) {
-				for (const Creature* tileCreature : *creatures) {
+				for (const auto& tileCreature : *creatures) {
 					if (!player->canWalkthrough(tileCreature)) {
 						return RETURNVALUE_NOTPOSSIBLE;
 					}
 				}
 			}
 
-			if (MagicField* field = getFieldItem()) {
+			if (auto field = getFieldItem()) {
```

Comment:
Could this be `const auto`
```suggestion
			if (const auto field = getFieldItem()) {
```